internal/security/proxy/token.go

```diff
@@ -19,34 +19,23 @@ package proxy
 
 import (
 	"encoding/json"
-	"io/ioutil"
+	"io"
 )
 
-type userTokenPair struct {
+type UserTokenPair struct {
 	User  string
 	Token string
 }
 
-type Writer interface {
-	Save(u string, t string) error
-}
-
-type TokenFileWriter struct {
-	Filename string
-}
-
-func (tf *TokenFileWriter) Save(u string, t string) error {
-
-	data := userTokenPair{
-		User:  u,
-		Token: t,
+func (ut *UserTokenPair) Save(w io.Writer) error {
```

Comment:
please see my previous comment about this function, because while we no longer have an interface for the Save method here, we do have this function which just calls json.MarshalIndent, w.Write and checks the error. Since this function is only ever called in one place, why not just call json.MarshalIndent and w.Write in that place? It would reduce the lines of code and the amount of error checking

Reply:
Reducing lines of code may not take priority currently. Please feel free to open an issue and submit fix for it.